File: python/http-internal-api/handler-external.py
import boto3
import json

lambdaClient = boto3.client('lambda', region_name = 'eu-west-2')

def handler(event, context):
    name = 'None Set'

    if 'queryStringParameters' in event and 'name' in event['queryStringParameters']:
        name = event['queryStringParameters']['name']

    response = lambdaClient.invoke(
        FunctionName = 'HttpInternalApiPython-dev-internalApi',
        InvocationType = 'RequestResponse',
        LogType = 'None',
        Payload = json.dumps({'name': name })
    )

    print(response)

    return {
        'statusCode': 200,
        'headers': {},
        'body': response['Payload'].read()
    }

File: js/http-internal-api/handler-external.js
const AWS = require('aws-sdk');

const lambda = new AWS.Lambda({
  apiVersion: 'latest',
  region: 'eu-west-2',
});

const params = {
  FunctionName: 'HttpInternalApiJs-dev-internalApi',
  InvocationType: 'RequestResponse',
  LogType: 'None',
  Payload: JSON.stringify({}),
};

exports.handler = (event, context, callback) => {
  const name = event.queryStringParameters && event.queryStringParameters.name
    ? event.queryStringParameters.name
    : 'None Set';

  params.Payload = JSON.stringify({
    name
  });

  console.log(params.Payload);

  const request = lambda.invoke(params);

  request
    .on('success', ({ data }) => {
      console.log({ name: 'success', data });
      callback(null, data.Payload);
    })
    .on('error', error => {
      console.log({ name: 'error', error });
      callback(error);
    })
    .on('complete', () => {
      console.log('DONE!!!');
    })
    .send()
  ;
}
